Zend/Optimizer/zend_optimizer.c

```diff
@@ -891,16 +891,23 @@ zend_function *zend_optimizer_get_called_func(
 					&op_array->scope->function_table, method_name);
 				if (fbc) {
 					bool is_private = (fbc->common.fn_flags & ZEND_ACC_PRIVATE) != 0;
-					bool is_final = (fbc->common.fn_flags & ZEND_ACC_FINAL) != 0;
-					bool same_scope = fbc->common.scope == op_array->scope;
 					if (is_private) {
 						/* Only use private method if in the same scope. We can't even use it
 						 * as a prototype, as it may be overridden with changed signature. */
+						bool same_scope = fbc->common.scope == op_array->scope;
 						return same_scope ? fbc : NULL;
 					}
 					/* If the method is non-final, it may be overridden,
 					 * but only with a compatible method signature. */
-					*is_prototype = !is_final;
+					/* We know it isn't a prototype if it can be overridden.
```

Comment:
```suggestion
					/* We know it is a prototype if it can be overridden.
```

And the above comment is somewhat outdated + partly superfluous

Reply:
Oops, my bad. I'll fix the comment and delete the old one.